hivemq-edge/src/main/java/com/hivemq/api/auth/provider/impl/simple/SimpleUsernameRolesProviderImpl.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.api.auth.provider.impl.simple;
+
+import com.hivemq.api.auth.provider.IUsernameRolesProvider;
+import com.hivemq.http.core.UsernamePasswordRoles;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * @author Simon L Johnson
+ */
+public class SimpleUsernameRolesProviderImpl implements IUsernameRolesProvider {
+
+    private final @NotNull Map<String, UsernamePasswordRoles> usernamePasswordMap;
+
+    public SimpleUsernameRolesProviderImpl() {
+        this.usernamePasswordMap = Collections.synchronizedMap(new HashMap<>());
```

Comment:
how about a ConcurrentHashMap instead,  optimised for reads? please change.

Reply:
Good point, fixed.